source/Halibut.Tests/ParallelRequestsFixture.cs

```diff
@@ -6,147 +6,266 @@
 using System.Threading;
 using System.Threading.Tasks;
 using FluentAssertions;
-using Halibut.Logging;
 using Halibut.Tests.Support;
 using Halibut.Tests.Support.TestAttributes;
 using Halibut.Tests.Support.TestCases;
+using Halibut.Tests.TestServices.Async;
 using Halibut.TestUtils.Contracts;
 using NUnit.Framework;
 
 namespace Halibut.Tests
 {
-    [Parallelizable(ParallelScope.Children)]
-    public class ParallelRequestsFixture : BaseTest
+    public static class ParallelRequestsFixture
     {
-        [Test]
-        [LatestAndPreviousClientAndServiceVersionsTestCases(testPolling: false, testWebSocket: false, testNetworkConditions: false,
-            testAsyncAndSyncClients: false // TODO - ASYNC ME UP!
-            )]
-        public async Task MultipleRequestsCanBeInFlightInParallel(ClientAndServiceTestCase clientAndServiceTestCase)
+        public class AsyncParallelRequestFixture : BaseTest
         {
-            using var clientAndService = await clientAndServiceTestCase.CreateTestCaseBuilder()
-                .WithStandardServices()
-                .Build(CancellationToken);
+            [Test]
+            [LatestAndPreviousClientAndServiceVersionsTestCases(testPolling: false, testWebSocket: false, testNetworkConditions: false, testAsyncAndSyncClients: false)]
```

Comment:
There was a `// TODO - ASYNC ME UP!` before. That's gone. Does this mean `testAsyncAndSyncClients` should be `true` now? or is it still not ready?

Reply:
Maybe more context on the PR description would have made sense.

The tests are written twice for sync and again for async.
The reason for this is if we try to use the async test proxy over a sync call we will cause thread pool exhaustion.
Also if we true to use threads in the async parallel tests we are not actually testing what we want to test since it should be the case we can without making threads run logs of parallel requests.

Eventually when everything is actually async this async side of the test should be changed to show that hundreds of concurrent requests can be made. Since everything is async, we wont have thread pool exhaustion.

Reply:
Ok, cool. 

So then, should `SendMessagesToTentacleInParallelAsync` also have `testAsyncAndSyncClients: false`?

Reply:
yes

Reply:
Done